### ISSUE TITLE
javax.annotation is not shaded

### ISSUE THREAD

**Issue by AlexandreCarlton** [launchdarkly/java-client#156]
In consuming the `launchdarkly-client` `jar`, I have noticed that it provides `javax.annotation` classes. This creates a few warnings when creating a shaded jar with Maven (which warns against its overlapping with [`jsr305`](https://mvnrepository.com/artifact/com.google.code.findbugs/jsr305)) and breaks the [`dependency:analyze`](https://maven.apache.org/plugins/maven-dependency-plugin/analyze-mojo.html) goal.

We have the following code:

https://github.com/launchdarkly/java-client/blob/4791702a70d4d70b8dcab4283131aa09652fa998/build.gradle#L224

so this is clearly deliberate. Could this decision be explained please? I am confused as to why this is the only unshaded package in the regular jar.

Thanks!

**Comment by eli-darkly** [comment]
Thanks for bringing this up. The presence of the JSR305 annotation classes is a legacy thing that I think at this point we should probably get rid of. I'll need to double-check to make sure there isn't some reason we can't do that, but otherwise it should be a pretty simple change.

Background: we don't actually use those annotations in the Java SDK, but we did use them in the public interface for the [`okhttp-eventsource`](https://github.com/launchdarkly/okhttp-eventsource) library which is used by the SDK. I _think_ the reason they were being embedded in the uberjar was that we had a somewhat unsophisticated method for building the jar: we could exclude a direct dependency, but we couldn't exclude an indirect dependency of something like `okhttp-eventsource` that we did want to embed. As for why they're not shaded—I think there is an inherent problem with shading annotation classes, in that any code that is looking for the annotation will expect them to have their original package names... although I'm not sure if it would matter in this case since the EventSource code is not visible.

But in any case, JSR305 was never really supported and [can cause problems for Java 9 developers](https://blog.codefx.org/java/jsr-305-java-9/), so it's probably best for us to remove it completely.
